pkg/apis/core/types.go

```diff
@@ -3356,9 +3360,21 @@ type PodStatus struct {
 	// give the resources on this node to a higher priority pod that is created after preemption.
 	// +optional
 	NominatedNodeName string
+
+	// HostIP holds the IP address of the host to which the pod is assigned. Empty if the pod has not started yet.
+	// A pod can be assigned to a node that has a problem in kubelet which in turns mean that HostIP will
+	// not be updated even if there is a node is assigned to pod
 	// +optional
 	HostIP string
```

Comment:
@thockin @liggitt  this is the part I didn't understand (I can't find the discussion) , when we did pluralize PodIPs the singular was obtained from the plural and removed from the type

Reply:
https://github.com/kubernetes/kubernetes/blob/a83cc51a19d1b5f2b2d3fb75574b04f587ec0054/pkg/apis/core/v1/conversion.go#L256-L287